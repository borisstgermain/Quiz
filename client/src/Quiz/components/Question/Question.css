.question {
    margin-bottom: 20px;
}

.title {
    margin-bottom: 30px;
}

.button {
    color: #421b75;
    border: none;
    background-color: #fcdb00;

    font-size: 18px;
    line-height: 26px;
}

.button:hover {
    background-color: #f5de5b;
}

.button--hide {
    position: absolute;

    visibility: hidden;
}

.answer {
    margin-bottom: 15px;

    font-size: 18px;
    font-weight: bold;
}

.answer--hide {
    visibility: hidden;
}

.answer--show {
    visibility: visible;
}

.comment {
    font-style: italic;
}

.comment--show {
    visibility: visible;
}

.comment--hide {
    visibility: hidden;
}
