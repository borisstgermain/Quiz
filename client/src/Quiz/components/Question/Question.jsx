import React from 'react';
import PropTypes from 'prop-types';

import style from './Question.css';

export default function Question({ question, answer, comments = '', isShown, clickHandler }) {
    return(
        <div className={style.question}>
            <p className={style.title}>{question}</p>
            <button className={isShown ? style['button--hide'] : style.button}
                onClick={clickHandler}
            >
                Показать ответ
            </button>
            <p className={`${style.answer}  ${isShown ? style['answer--show'] : style['answer--hide']}`}>
                {answer}
            </p>
            <p className={`${style.comment}  ${isShown ? style['comment--show'] : style['comment--hide']}`}>
                {comments}
            </p>
        </div>
    )
}

Question.propTypes = {
    question: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired,
    comment: PropTypes.string,
    isShown: PropTypes.bool.isRequired,
    clickHandler: PropTypes.func.isRequired
};
