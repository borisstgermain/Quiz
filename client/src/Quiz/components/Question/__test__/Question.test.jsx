import React from 'react';
import { shallow } from 'enzyme';
import Question from '../Question';

describe('Question component', () => {
  describe('Smoke tests', () => {
    const props = {
      question: 'What is this?',
      answer: 'JS',
      comment: '',
      clickHandler: jest.fn(),
      isShown: false,
    };

    test('Render without crashing', () => {
      shallow(<Question {...props} />);
    });
  });

  describe('Component testing', () => {
    let wrapper = null;
    let props = null;

    beforeEach(() => {
      props = {
        question: 'What is this?',
        answer: 'JS',
        comment: '',
        clickHandler: jest.fn(bool => !bool),
        isShown: false,
      };
      wrapper = shallow(<Question {...props} />);
    });

    test('Contains elements, correct text / Snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('Correct props', () => {
      expect(wrapper.instance().props).toEqual(props);
    });

    test('Click handler changes prop â€“ isShow ', () => {
      expect(wrapper).toMatchSnapshot();

      props.isShown = wrapper.instance().props.clickHandler(props.isShown);
      const wrapperAfterClick = shallow(<Question {...props} />);

      expect(wrapperAfterClick).toMatchSnapshot();
      expect(wrapperAfterClick.instance().props.isShown).toEqual(true);
    });
  });
});
