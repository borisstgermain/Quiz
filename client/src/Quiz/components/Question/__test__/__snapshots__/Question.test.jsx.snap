// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component Component testing Click handler changes prop – isShow  1`] = `
<div
  className="question"
>
  <p
    className="title"
  >
    What is this?
  </p>
  <button
    className="button"
    onClick={[Function]}
  >
    Показать ответ
  </button>
  <p
    className="answer  answer--hide"
  >
    JS
  </p>
  <p
    className="comment  comment--hide"
  />
</div>
`;

exports[`Question component Component testing Click handler changes prop – isShow  2`] = `
<div
  className="question"
>
  <p
    className="title"
  >
    What is this?
  </p>
  <button
    className="button--hide"
    onClick={[Function]}
  >
    Показать ответ
  </button>
  <p
    className="answer  answer--show"
  >
    JS
  </p>
  <p
    className="comment  comment--show"
  />
</div>
`;

exports[`Question component Component testing Contains elements, correct text / Snapshot 1`] = `
<div
  className="question"
>
  <p
    className="title"
  >
    What is this?
  </p>
  <button
    className="button"
    onClick={[Function]}
  >
    Показать ответ
  </button>
  <p
    className="answer  answer--hide"
  >
    JS
  </p>
  <p
    className="comment  comment--hide"
  />
</div>
`;
